README.md

```diff
@@ -38,229 +88,412 @@
-                        esy shell.
+8. For more options:
+    ```
+    % esy help
+    ```
+
+### Configuring Your `package.json`
+
+`esy` knows how to build your package and its dependencies by looking at the
+`esy` config section in your `package.json`.
+
+This is how it looks for a [jbuilder][] based project:
 
 ```
+{
+  "name": "example-package",
+  "version": "1.0.0",
 
-## Try An Example
+  "esy": {
+    "build": [
+      "jbuilder build bin/hello.exe"
+    ],
+    "install": [
+      "jbuilder build @install",
+      "jbuilder install --prefix=$cur__install"
+    ],
+    "buildsinsource": "_build"
+  },
 
+  "dependencies": {
+    "anotherpackage": "1.0.0"
+  }
+}
 ```
-# Make sure esy is installed
-npm install -g esy
 
-# Clone the example esy project
-git clone git@github.com:esy-ocaml/esy-ocaml-project.git
+#### Specify Build & Install Commands
 
-cd esy-ocaml-project
+The crucial pieces of configuration are `esy.build` and `esy.install` keys, then
+specify how to build and install built artifacts.
 
-# Install project dependencies
-esy install
+##### `esy.build`
 
-# Build project
-esy build
+Describe how your project's default targets should be built by specifying
+a list of commands with `esy.build` config key.
 
-# Now run some commands inside the project's environment
-esy ./_install/bin/hello.native
+For example for a [jbuilder][] based project you'd want to call `jbuilder build`
+command for each of the target which is going to be installed later:
 
-# Shell into project's environment
-esy shell
 ```
+{
+  "esy": {
+    "build": [
+      "jbuilder build bin/hello.exe",
+      "jbuilder build lib/MyLib.cmxa",
+    ]
+  }
+}
+```
+
+Commands specified in `esy.build` are always executed for the root's project
+when user calls `esy build` command.
+
+##### `esy.install`
 
-## Enjoy The Cache
+Describe how you project's built artifacts should be installed by specifying a
+list of commands with `esy.install` config key.
+
+For example for a [jbuilder][] based project you'd want to generate an `.install`
+file and then call `jbuild install` command:
 
-The previous example may have taken 10 minutes to build, but with `esy`'s
-cache, the second time will be instant.
 
 ```
-rm -rf node_modules
-esy install
-esy build
+{
+  "esy": {
+    "build": [...],
+    "install": [
+      "jbuilder build @install",
+      "jbuilder install --prefix $cur__install",
+    ]
+  }
+}
 ```
 
-`esy`'s cache is package-granular and takes into account anything that could
-influence the build. If you create another project with 90% of the same
-dependencies, there's a good chance that 90% of the build is performed
-instantly (pulled from cache).
+Note the `$cur__install` variable which is used for an installation prefix. This
+variable is a part of [build environment](#build-environment) provided by Esy.
 
+#### Enforcing Out Of Source Builds
 
-## Configuring Your `package.json`
+Esy requires packages to be built "out of source".
```

Comment:
Is it correct to say that it "requires" packages to be built "out of source"? I think it's more like "Esy projects that build into specific, known locations (such as `_build`) are able to take advantage of many optimizations, and provide the best integration with IDE tooling"

(Note, merlin really sucks when your packages are first copied into another unknown location before the package build takes place, so I might get why you'd say it's "required" that packages build into _build or configurable arbitrary location).

Reply:
I'll leave this as-is right now. You are correct that it's not just "out of source" though.